akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ProtoAnySerialization.scala

```diff
@@ -177,6 +214,50 @@ import scalapb.options.Scalapb
     }
   }
 
+  def toSerializedEvent(scalaPbAny: ScalaPbAny): Option[SerializedEvent] = {
+    // see corresponding typeUrl cases in `deserialize`
+
+    val typeUrl = scalaPbAny.typeUrl
+    if (typeUrl == ProtoAnyTypeUrl) {
+      // We don't try to optimize this case. One level of indirection too much, and probably not a common case.
+      None
+    } else if (typeUrl.startsWith(GoogleTypeUrlPrefix)) {
+      val messageClass = resolveTypeUrl(typeUrl).messageClass
+      val serializer = serialization.serializerFor(messageClass)
```

Comment:
We could potentially resolve the serializer up front/once in the ResolvedType instead of for each message.